compositor_pipeline/src/queue/internal_queue.rs

```diff
@@ -131,4 +134,22 @@ impl InternalQueue {
     pub fn get_next_output_buffer_pts(&self) -> Duration {
         Duration::from_secs_f64(self.sent_batches_counter as f64 / self.output_framerate.0 as f64)
     }
+
+    pub fn register_input_listener(
+        &mut self,
+        input_id: InputId,
+        callback: Box<dyn FnOnce() + Send>,
+    ) {
+        self.inputs_listeners
+            .entry(input_id)
+            .or_insert_with(std::vec::Vec::new)
+            .push(callback)
+    }
+
+    fn call_input_listeners(&mut self, input_id: &InputId) {
```

Comment:
[NIT]
Since we use terminology from the observer pattern (`listener`), consider using it here as well:
`call` -> `notify`
`register` -> `subscribe`

Reply:
no objection to the subscribe, but notify implies async action which is not true in this case